support-frontend/app/controllers/CreateSubscriptionController.scala

```diff
@@ -382,3 +396,45 @@ class CreateSubscriptionController(
     )
   }
 }
+
+object ThreeTierTest {
+  def augmentPromoCode(body: CreateSupportWorkersRequest) = {
+    val isGuardianWeekly = body.product match {
+      case _: GuardianWeekly => true
+      case _ => false
+    }
+    val isInThreeTier = body.threeTierCreateSupporterPlusSubscription.getOrElse(false)
+
+    val threeTierPromoCode = if (isInThreeTier && isGuardianWeekly) {
+      val maybeId = CountryGroup.byCountryCode(body.billingAddress.country.alpha2).map(_.id)
+
+      val maybeBillingPeriod = body.product.billingPeriod match {
+        case Monthly => Some("monthly")
+        case Annual => Some("annual")
+        case _ => None
+      }
+
+      (for {
+        id <- maybeId
+        billingPeriod <- maybeBillingPeriod
+      } yield Map(
+        "uk-monthly" -> "3TIER_WEEKLY_UK_MONTHLY",
+        "uk-annual" -> "3TIER_WEEKLY_UK_ANNUAL",
+        "eu-monthly" -> "3TIER_WEEKLY_EU_MONTHLY",
+        "eu-annual" -> "3TIER_WEEKLY_EU_ANNUAL",
+        "int-monthly" -> "3TIER_WEEKLY_INT_MONTHLY",
+        "int-annual" -> "3TIER_WEEKLY_INT_ANNUAL",
+        "us-monthly" -> "3TIER_WEEKLY_US_MONTHLY",
+        "us-annual" -> "3TIER_WEEKLY_US_ANNUAL",
+        "ca-monthly" -> "3TIER_WEEKLY_CA_MONTHLY",
+        "ca-annual" -> "3TIER_WEEKLY_CA_ANNUAL",
+        "nz-monthly" -> "3TIER_WEEKLY_NZ_MONTHLY",
+        "nz-annual" -> "3TIER_WEEKLY_NZ_ANNUAL",
+        "au-monthly" -> "3TIER_WEEKLY_AU_MONTHLY",
+        "au-annual" -> "3TIER_WEEKLY_AU_ANNUAL",
+      ) get s"$id-$billingPeriod").flatten
```

Comment:
let's bomb out if we didn't find a promo code - we don't want to accidentally sell non discounted ones
```suggestion
      Some((for {
        id <- maybeId
        billingPeriod <- maybeBillingPeriod
      } yield Map(
        "uk-monthly" -> "3TIER_WEEKLY_UK_MONTHLY",
        "uk-annual" -> "3TIER_WEEKLY_UK_ANNUAL",
        "eu-monthly" -> "3TIER_WEEKLY_EU_MONTHLY",
        "eu-annual" -> "3TIER_WEEKLY_EU_ANNUAL",
        "int-monthly" -> "3TIER_WEEKLY_INT_MONTHLY",
        "int-annual" -> "3TIER_WEEKLY_INT_ANNUAL",
        "us-monthly" -> "3TIER_WEEKLY_US_MONTHLY",
        "us-annual" -> "3TIER_WEEKLY_US_ANNUAL",
        "ca-monthly" -> "3TIER_WEEKLY_CA_MONTHLY",
        "ca-annual" -> "3TIER_WEEKLY_CA_ANNUAL",
        "nz-monthly" -> "3TIER_WEEKLY_NZ_MONTHLY",
        "nz-annual" -> "3TIER_WEEKLY_NZ_ANNUAL",
        "au-monthly" -> "3TIER_WEEKLY_AU_MONTHLY",
        "au-annual" -> "3TIER_WEEKLY_AU_ANNUAL",
      ) get s"$id-$billingPeriod").flatten.get)
```